packages/lesswrong/components/notifications/NotificationsPage/notificationsPageTabs.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { FC, ReactNode, createContext, useCallback, useContext, useState } from "react";
+
+export const notificationPageTabs = [
+  {
+    name: "all",
+    type: undefined,
+  },
+  {
+    name: "comments",
+    type: "newComment",
+  },
+  {
+    name: "new posts",
+    type: "newPost",
+  },
+] as const;
+
+export type NotificationsPageTab = typeof notificationPageTabs[number];
+
+export type NotificationsPageTabName = NotificationsPageTab["name"];
+
+export const isNotificationsPageTabName = (
+  name: string,
+): name is NotificationsPageTabName => {
+  for (const tab of notificationPageTabs) {
+    if (tab.name === name) {
+      return true;
+    }
+  }
+  return false;
+}
+
+type NotificationsPageTabContext = {
+  tab: NotificationsPageTab,
+  setTab: (tab: NotificationsPageTabName) => void,
+}
+
+const notificationsPageTabContext = createContext<NotificationsPageTabContext>({
+  tab: notificationPageTabs[0],
+  setTab: () => {},
+});
+
+export const NotificationsPageTabContextProvider: FC<{
```

Comment:
What's the purpose for using context here? It seems like the notifications page tab is only used and updated in one component anyway (`NotificationsPageFeed`, which also passes the tab name down to a child via a prop)

Reply:
It used to be accessed in multiple components, but isn't anymore. Doesn't seem worth changing now.